synapse/lib/stormtypes.py

```diff
@@ -5165,6 +5241,9 @@ def __init__(self, runt, tdef):
         self.locls.update(self.getObjLocals())
         self.locls['iden'] = self.valu.get('iden')
 
+    def __hash__(self):
+        return hash(self.locls['iden'])
```

Comment:
this should hash the tuple of typename and iden since iden could be externally controlled/created

Reply:
So I actually initially did this, but it ends up making things like `s_stormtypes.Str("foo")` not equal to just the plain string of `foo`. I kinda wanted to make those line up, but avoiding other collisions is also nice. 

But in this particular case (and for strings too), I kinda have to reach around the type system by punching variables in using opts to make this even a problem (see https://github.com/vertexproject/synapse/pull/2225/files#diff-5c10f002edaca1b0cbcbc511441d1edd6e4047a6e5afdf0d32c17751f17d61bfR1213) . But it's still fairly easy to do so. 

TL;DR: Is it cool for `s_stormtypes.Str("foo")` to not equal `foo` until we have full stormtypes everywhere?